src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -0,0 +1,197 @@
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.SUCCESS;
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.UNRECOVERABLE_FAILURE;
+
+
+@Service
+@Profile("!nosb") // do not register for the nosb (test) profile
+public class PaymentMessageProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(PaymentMessageProcessor.class);
+
+    private final PaymentMessageHandler paymentMessageHandler;
+    private final IMessageReceiver messageReceiver;
+    private final int maxDeliveryCount;
+
+    public PaymentMessageProcessor(PaymentMessageHandler paymentMessageHandler,
+                                   IMessageReceiver messageReceiver,
+                                   @Value("${azure.servicebus.payments.max-delivery-count}") int maxDeliveryCount) {
+        this.paymentMessageHandler = paymentMessageHandler;
+        this.messageReceiver = messageReceiver;
+        this.maxDeliveryCount = maxDeliveryCount;
+    }
+
+
+    /**
+     * Reads and processes next message from the queue.
+     *
+     * @return false if there was no message to process. Otherwise true.
+     */
+    public boolean processNextMessage() throws ServiceBusException, InterruptedException {
+        IMessage message = messageReceiver.receive();
+        if (message != null) {
+            MessageProcessingResult result = process(message);
+            tryFinaliseProcessedMessage(message, result);
+        }
+        return message != null;
+    }
+
+    private MessageProcessingResult process(IMessage message) {
+        log.info("Started processing message with ID {}", message.getMessageId());
+
+        PaymentMessage payment = null;
+
+        try {
+            payment = PaymentMessageParser.parse(message.getMessageBody());
+            logMessageParsed(message, payment);
+            paymentMessageHandler.handlePaymentMessage(payment);
+            log.info("Processed message with ID {}. Envelope ID: {}", message.getMessageId(), payment.envelopeId);
+            return new MessageProcessingResult(SUCCESS);
+        } catch (InvalidMessageException ex) {
+            log.error("Rejected message with ID {}, because it's invalid", message.getMessageId(), ex);
+            return new MessageProcessingResult(UNRECOVERABLE_FAILURE, ex);
+        } catch (PayHubClientException ex) {
+            logMessageProcessingError(message, payment, ex);
+            if (ex.getStatus() == HttpStatus.BAD_REQUEST.CONFLICT) {
+                return new MessageProcessingResult(UNRECOVERABLE_FAILURE, ex);
+            }
+        } catch (Exception ex) {
+            logMessageProcessingError(message, payment, ex);
+        }
+
+        return new MessageProcessingResult(POTENTIALLY_RECOVERABLE_FAILURE);
```

Comment:
My preference is the each catch clause returns a result, as in my opinion it reads a bit easier, but that's just a suggestion.